### PR TITLE
add blank_penalty for offline transducer

### PR DESCRIPTION
**Pull request by chiiyeh**
Fix https://github.com/k2-fsa/sherpa-onnx/issues/541
Tested with `sherpa-onnx-offline`. High penalty will force out more words.

### CODE REVIEW

File: sherpa-onnx/csrc/offline-transducer-greedy-search-decoder.cc [hunk 1]
```diff
@@ -46,7 +46,10 @@ OfflineTransducerGreedySearchDecoder::Decode(Ort::Value encoder_out,
     start += n;
     Ort::Value logit = model_->RunJoiner(std::move(cur_encoder_out),
                                          std::move(cur_decoder_out));
-    const float *p_logit = logit.GetTensorData<float>();
+    float *p_logit = logit.GetTensorMutableData<float>();
+    if (blank_penalty_ > 0.0) {
+      p_logit[0] -= blank_penalty_; // assuming blank id is 0
```
Comment (csukuangfj) [inline sherpa-onnx/csrc/offline-transducer-greedy-search-decoder.cc]: Shall we also consider the case when `batch_size > 1` ?

`p_logit[0]` is only for the first utterance.

 We need to process
```bash
p_logit[vocab_size*i + 0]
// for in range(n)
```

You can move this `if` statement into the `for` statement below.

File: python-api-examples/non_streaming_server.py [hunk 2]
```diff
@@ -862,6 +876,7 @@ def create_recognizer(args) -> sherpa_onnx.OfflineRecognizer:
             max_active_paths=args.max_active_paths,
             hotwords_file=args.hotwords_file,
             hotwords_score=args.hotwords_score,
+            blank_penalty=args.blank_penalty,
```
Comment (csukuangfj) [inline python-api-examples/non_streaming_server.py:879]: Could you also update
https://github.com/k2-fsa/sherpa-onnx/blob/b9291246e86f6f4aff68449cd5a228feb2d6695d/sherpa-onnx/python/sherpa_onnx/offline_recognizer.py#L38
to add an extra argument
```
blank_penalty: float = 0
```
and also add docstring for it?

Note that you need to pass the argument to
https://github.com/k2-fsa/sherpa-onnx/blob/b9291246e86f6f4aff68449cd5a228feb2d6695d/sherpa-onnx/python/sherpa_onnx/offline_recognizer.py#L114

**Review by csukuangfj — COMMENTED** [review]
Thanks! Left some minor comments. Otherwise, it looks good to me.

**Comment by csukuangfj** [comment]
By the way, please merge the master branch into your current branch so the CI can pass.

File: sherpa-onnx/csrc/offline-transducer-greedy-search-decoder.cc (hunk 1 above)
Comment (chiiyeh) [reply sherpa-onnx/csrc/offline-transducer-greedy-search-decoder.cc]: Fixed!

File: python-api-examples/non_streaming_server.py (hunk 2 above)
Comment (chiiyeh) [reply python-api-examples/non_streaming_server.py:879]: Fixed! 
I also noticed there is no docstring for the hotwords params, as well as the max_active_paths was not propagated into the `OfflineRecognizerConfig`. Should I raise two seperate PR to fix these?

Comment (csukuangfj) [reply python-api-examples/non_streaming_server.py:879]: > Should I raise two seperate PR to fix these?

Yes, please do that in a separate PR. Thanks!

**Comment by csukuangfj** [comment]
Could you fix the c++ code style check?
https://github.com/k2-fsa/sherpa-onnx/actions/runs/7648706009/job/20842763975#step:4:38
```
Check last commit
sherpa-onnx/csrc/math.h:100:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
Done processing sherpa-onnx/csrc/math.h
Total errors found: 1
[FAILED] sherpa-onnx/csrc/math.h
Error: Process completed with exit code 1
```

You can run
```bash
pip install clang-format
cd /path/to/sherpa-onnx
./scripts/check_style_cpplint.sh
./scripts/check_style_cpplint.sh 1
./scripts/check_style_cpplint.sh 2
```
to check the style locally.

**Comment by csukuangfj** [comment]
Thank you for your first-time contribution!

**Pull request merged**